crates/sui-storage/src/indexes.rs

```diff
@@ -1328,10 +1328,7 @@ impl IndexStore {
                 )
             })
             .await
-            .unwrap()
-            .map_err(|e| {
-                SuiError::ExecutionError(format!("Failed to read balance frm DB: {:?}", e))
```

Comment:
The underlying error is a `TypedStoreError` which is already a `SuiError`, so we should just return that